File: regular_expressions/pl/processing_emails.pl
#!/usr/bin/perl

######################################################
## using PL regular expressions to find and
## retrieve the components of an email address: 1)
## user name; 2) host name; 3) host extension.
## @author: vladimir kulyukin
######################################################

use strict;
use warnings;

my $TXT_01 = '
John   Balbiro	A1001	john.balbiro@usu.edu
Alice  Nelson	A0011	alice.nelson@workflow.net
Jacob  Roberts	A1100	j.s.roberts@gmail.com
';

my $email_pat_1 = '([\w.-]+@[\w.-]+\.(?:com|net|org|edu))';
my @emails_1 = ();
while ( $TXT_01 =~ m/$email_pat_1/g ) {
    push(@emails_1, $1);
}

my $email_pat_2 = '([\w.-]+@[\w.-]+)\.(com|net|org|edu)';
my @emails_2 = ();
while ( $TXT_01 =~ m/$email_pat_2/g ) {
    my @lst = ();
    push(@lst, $1);
    push(@lst, $2);
    push(@emails_2, \@lst);
}

my $email_pat_3 = '([\w.-]+)@([\w.-]+)\.(com|net|org|edu)';
my @emails_3 = ();
while ( $TXT_01 =~ m/$email_pat_3/g ) {
    my @lst = ();
    push(@lst, $1);
    push(@lst, $2);
    push(@lst, $3);
    push(@emails_3, \@lst);
}

print "$_\n"    foreach(@emails_1);
print "@{$_}\n" foreach(@emails_2);
print "@{$_}\n" foreach(@emails_3);

File: regular_expressions/py/processing_emails.py
#!/usr/bin/python

######################################################
## using Python regular expressions to find and
## retrieve the components of an email address: 1)
## user name; 2) host name; 3) host extension.
## @author: vladimir kulyukin
######################################################

import re

TXT_01 = '\n\
John   Balbiro	A1001	john.balbiro@usu.edu\n\
Alice  Nelson	A0011	alice.nelson@workflow.net\n\
Jacob  Roberts	A1100	j.s.roberts@gmail.com\n\
'

## we do not want to extract com|net|org|edu as a separate
## group, so we add ?: after the left parenthesis. this set of
## parentheses will not count as a group result
## 1 group
email_pat_1 = r'([\w.-]+@[\w.-]+\.(?:com|net|org|edu))'
emails_1 = re.findall(email_pat_1,  TXT_01)
print emails_1

## 2 groups
email_pat_2 = r'([\w.-]+@[\w.-]+)\.(com|net|org|edu)'
emails_2 = re.findall(email_pat_2, TXT_01)
print emails_2

## 3 groups
email_pat_3 = r'([\w.-]+)@([\w.-]+)\.(com|net|org|edu)'
emails_3 = re.findall(email_pat_3, TXT_01)
print emails_3
